lib/install.go

```diff
@@ -138,97 +136,78 @@ func install(product Product, tfversion string, binPath string, installPath stri
 	/* set symlink to desired version */
 	CreateSymlink(installFileVersionPath, binPath)
 	logger.Infof("Switched terraform to version %q", tfversion)
-	addRecent(product, tfversion, installPath) //add to recent file for faster lookup
+	addRecentVersion(product, tfversion, installPath) //add to recent file for faster lookup
 	return
 }
 
-// addRecent : add to recent file
-func addRecent(product Product, requestedVersion string, installPath string) {
+// getRecentFileName : Obtain recent version filename for product
+func getRecentFileName(product Product) string {
+	return recentFilePrefix + product.GetId() + ".json"
+}
 
+// addRecent : add to recent file
+func addRecentVersion(product Product, requestedVersion string, installPath string) {
 	installLocation = GetInstallLocation(installPath) //get installation location -  this is where we will put our terraform binary file
 	recentFilePath := filepath.Join(installLocation, getRecentFileName(product))
 
-	fileExist := CheckFileExist(recentFilePath)
-	if fileExist {
-		lines, errRead := ReadLines(recentFilePath)
+	// Obtain pre-existing latest version
+	versions := getRecentVersions(product, installPath)
 
-		if errRead != nil {
-			logger.Errorf("Error reading %q file: %v", recentFilePath, errRead)
-			return
-		}
-
-		for _, line := range lines {
-			if !validVersionFormat(line) {
-				logger.Infof("File %q is dirty (recreating cache file)", recentFilePath)
-				RemoveFiles(recentFilePath)
-				CreateRecentFile(requestedVersion, installPath, recentFilePath)
-				return
-			}
+	// Check for requestedVersion in versions list
+	for versionIndex, versionVal := range versions {
+		if versionVal == requestedVersion {
+			versions = append(versions[:versionIndex], versions[versionIndex+1:]...)
 		}
+	}
 
-		versionExist := versionExist(requestedVersion, lines)
+	// Add new version to start of slice
+	versions = append([]string{requestedVersion}, versions...)
+	if len(versions) > 3 {
+		versions = versions[0:2]
+	}
 
-		// @TODO Does this not duplicate the behavoir of CreateRecentFile, called above? (possibly )
-		if !versionExist {
-			if len(lines) >= 3 {
-				_, lines = lines[len(lines)-1], lines[:len(lines)-1]
+	// Write new versions back to recent files
+	recentVersionFh, err := os.OpenFile(recentFilePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
```

Comment:
I'm one of those nerds who has custom umask set to not allow group and others by default. I'm from minority of users though. But I still would prefer file creation operations to take into account `umask` value that I define and fallback to sane default of `0644` otherwise.

Reply:
As mentioned in the previous PR, I suspect this is the case?
Based on: `it is created with mode perm (before umask)` (https://pkg.go.dev/github.com/pborman/getopt#section-readme)

Reply:
Yep, the other thread resolved my doubts. Thanks. I'm resolving this thread.